hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongService.java

```diff
@@ -147,4 +165,99 @@ private void clearPartitionReplica(int partitionId) {
             }
         }
     }
+
+    @Override
+    public Runnable prepareMergeRunnable() {
+        IPartitionService partitionService = nodeEngine.getPartitionService();
+        Map<Integer, List<AtomicLongContainer>> containerMap = createHashMap(partitionService.getPartitionCount());
+        Address thisAddress = nodeEngine.getClusterService().getThisAddress();
+
+        for (Map.Entry<String, AtomicLongContainer> entry : containers.entrySet()) {
+            AtomicLongContainer container = entry.getValue();
+            if (!(getMergePolicy(container) instanceof DiscardMergePolicy)) {
+                String name = entry.getKey();
+                int partitionId = partitionService.getPartitionId(StringPartitioningStrategy.getPartitionKey(name));
+                if (thisAddress.equals(partitionService.getPartitionOwner(partitionId))) {
+                    // add your owned values to the map so they will be merged
+                    List<AtomicLongContainer> containerList = containerMap.get(partitionId);
+                    if (containerList == null) {
+                        containerList = new ArrayList<AtomicLongContainer>(containers.size());
+                        containerMap.put(partitionId, containerList);
+                    }
+                    containerList.add(container);
+                }
+            }
+        }
+        containers.clear();
+
+        return new Merger(containerMap);
+    }
+
+    private SplitBrainMergePolicy getMergePolicy(AtomicLongContainer container) {
+        String mergePolicyName = container.getConfig().getMergePolicyConfig().getPolicy();
+        return mergePolicyProvider.getMergePolicy(mergePolicyName);
+    }
+
+    private class Merger implements Runnable {
+
+        private static final int TIMEOUT_FACTOR = 500;
+
+        private final Map<Integer, List<AtomicLongContainer>> containerMap;
+
+        Merger(Map<Integer, List<AtomicLongContainer>> containerMap) {
+            this.containerMap = containerMap;
+        }
+
+        @Override
+        public void run() {
+            // we cannot merge into a 3.9 cluster, since not all members may understand the QueueMergeOperation
+            if (nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V3_10)) {
+                return;
+            }
+
+            final ILogger logger = nodeEngine.getLogger(CollectionService.class);
```

Comment:
You can use the `AtomicLongService.class` to be more specific

Reply:
Now you know for sure, where I copied the code from :D